language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/MakeTypePublicCodeAction.java

```diff
@@ -0,0 +1,114 @@
+import org.ballerinalang.langserver.commons.codeaction.spi.DiagBasedPositionDetails;
+import org.eclipse.lsp4j.*;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code Action to make type public.
+ *
+ * @since 2201.0.3
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class MakeTypePublicCodeAction extends AbstractCodeActionProvider {
+    public static final String NAME = "MakeTypePublicCodeAction";
+    public static final String DIAGNOSTIC_CODE = "BCE2038";
+
+    @Override
+    public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
+                                                    DiagBasedPositionDetails positionDetails,
+                                                    CodeActionContext context) {
+        if ((!DIAGNOSTIC_CODE.contains(diagnostic.diagnosticInfo().code())) || context.currentSyntaxTree().isEmpty()
+                || context.currentSemanticModel().isEmpty()) {
+            return Collections.emptyList();
+        }
+        Range diagnosticRange = new Range(CommonUtil.toPosition(diagnostic.location().lineRange().startLine()),
+                CommonUtil.toPosition(diagnostic.location().lineRange().endLine()));
+        NonTerminalNode nonTerminalNode = CommonUtil.findNode(diagnosticRange, context.currentSyntaxTree().get());
+        Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(nonTerminalNode);
+        if (symbol.isEmpty() || symbol.get().getModule().isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ModuleID moduleID = symbol.get().getModule().get().id();
+        String orgName = moduleID.orgName();
+        String moduleName = moduleID.moduleName();
+        Optional<Project> project = context.workspace().project(context.filePath());
+        if (project.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Optional<Path> filePath = CommonUtil.getFilePathForDependency(orgName, moduleName, project.get(),
+                symbol.get(), context);
+        if (filePath.isEmpty() || context.workspace().syntaxTree(filePath.get()).isEmpty()) {
+            return Collections.emptyList();
+        }
+        URI uri = filePath.get().toUri();
+
+        Optional<NonTerminalNode> node = CommonUtil.findNode(symbol.get(),
+                context.workspace().syntaxTree(filePath.get()).get());
+        if (node.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Position startPosition = new Position();
+
+        if (node.get().kind().equals(SyntaxKind.TYPE_DEFINITION)) {
+            TypeDefinitionNode typeDefinitionNode = (TypeDefinitionNode) node.get();
+            startPosition = CommonUtil.toPosition(typeDefinitionNode.typeKeyword().lineRange().startLine());
+        }
+        if (node.get().kind().equals(SyntaxKind.CLASS_DEFINITION)) {
```

Comment:
If we can get `node.get().kind()` to a separate variable, we can avoid repeating this calculation twice